src/DistributedLock.Etcd/EtcdLeaseDistributedLockProvider.cs

```diff
@@ -0,0 +1,29 @@
+using dotnet_etcd.interfaces;
+
+namespace Medallion.Threading.Etcd;
+
+/// <summary>
+/// Implements <see cref="IDistributedLockProvider"/> for <see cref="AzureBlobLeaseDistributedLock"/>
+/// </summary>
+public sealed class EtcdLeaseDistributedLockProvider : IDistributedLockProvider
+{
+    private readonly IEtcdClient _blobContainerClient;
+    private readonly Action<EtcdLeaseOptionsBuilder>? _options;
+
+    /// <summary>
+    /// Constructs a provider that scopes blobs within the provided <paramref name="blobContainerClient"/> and uses the provided <paramref name="options"/>.
+    /// </summary>
+    public EtcdLeaseDistributedLockProvider(IEtcdClient blobContainerClient,

```

Comment:
Probably blobContainerClient here should always be `etcdClient` / `_etcdClient`?